src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -220,6 +236,33 @@ private void getEvents(HttpServletRequest request, HttpServletResponse response,
     return;
   }
 
+  private boolean joinEvent(HttpServletRequest request, HttpServletResponse response, UserService userService, String eventId)
+      throws IOException {
+    String currentUserId = null;
+    try {
+      currentUserId = userService.getCurrentUser().getUserId();
+    } catch (Exception e) {
+      System.err.println("Can't get current user id: " + e);
+      response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      return false;
+    }
+
+    if (!EventStorage.userHasAccessToEvent(currentUserId, eventId)) {
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return false;
+    }
+    
+    try {
+      UserStorage.joinEvent(currentUserId, eventId);
+    } catch (Exception e) {
```

Comment:
Catching a general Exception is usually bad practice. What kind of Exception are you expecting here?

Reply:
How can I understand it?

Reply:
`catch (Exception e)` is catching any exception that can happen but joinEvent probably only throws a derived type of Exception. Generally, you should catch the narrowest Exception type as possible. 
You can catch any Exception if you want to fail gracefully. But as far as I know, exception in servlet are already mapped to INTERNAL_ERROR responses so you don't need to catch it here.

Reply:
Maybe add a TODO here, it is not required change but nice to have.